standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -1299,7 +1297,9 @@ private void create_database_core(RawStore ms, final Database db)
       success = ms.commitTransaction();
     } finally {
       if (!success) {
-        ms.rollbackTransaction();
+        if (openTransaction) {
```

Comment:
why are you checking for an `openTransaction` flag here? I don't see the same check in other places. 
````
if (isActiveTransaction() && transactionStatus != TXN_STATUS.ROLLBACK) {
    currentTransaction.rollback();
}
````

Reply:
In `create_database_core` it may throw exception before calling `openTransaction()`, so it may not need call `rollbackTransaction()` in finally block.